tests/IntMapValidity.hs

```diff
@@ -0,0 +1,52 @@
+module IntMapValidity (valid) where
+
+import Data.Bits (xor, (.&.))
+import Data.IntMap.Internal
+import Test.QuickCheck (Property, property, (.&&.))
+
+{--------------------------------------------------------------------
+  Assertions
+--------------------------------------------------------------------}
+-- | Returns true iff the internal structure of the IntMap is valid.
+valid :: IntMap a -> Property
+valid t
+  = property (nilNeverChildOfBin t) .&&.
+    property (commonPrefix t) .&&.
+    property (maskRespected t)
```

Comment:
I think we probably want to use `counterexample` rather than `property` here, to label each type of failure. I wonder if we should push the `Property` business into `commonPrefix` and `maskRespected` so we can see where in the tree the failure lies. That probably shouldn't be necessary for `nilNeverChildOfBin`, since that will be visually obvious.

Reply:
Switched to using counterexample for each of these. It would take more work to push `Property` into the other functions and be done in such a way that we would get actionable information from it, so if its alright with you I'd like to hold off on that for now.